sdk/identity/azure-identity/azure/identity/_internal/aad_client_base.py

```diff
@@ -85,6 +102,7 @@ def _build_pipeline(self, config=None, policies=None, transport=None, **kwargs):
 
     def _process_response(self, response, request_time):
         # type: (PipelineResponse, int) -> AccessToken
+        self._last_refresh_time = int(time.time())   # no matter succeed or not, update the last refresh time
```

Comment:
This line executes after the client has received a response, so the current time will be somewhat distant from the time the request was sent. You could just use `request_time`:
```suggestion
        self._last_refresh_time = request_time   # no matter succeed or not, update the last refresh time
```